### PR TITLE
Create GET /api/articles endpoint

### PR DESCRIPTION
**Pull request by AkivaKn**
Build endpoint using TDD and update endpoints.json No error handling necessary in this instance.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,84 +3,111 @@ const app = require("../db/app");
...
+describe("/api/articles", () => {
+  describe("GET", () => {
+    test("GET 200: Responds with an array of article objects sorted by date, with a comment_count key and no body key", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles.length).toBe(13);
+          expect(articles[0].comment_count).toBe(2);
+          expect(articles).toBeSortedBy("created_at", { descending: true });
```
Comment (pippaamy) [inline __tests__/app.test.js:98]: I would have sorted by in a separate test

File: db/app.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
+const {
+  getArticleById,
+  getArticles,
+} = require("./controllers/articles-controller");
 
-const app = express()
+const app = express();
 
-app.use(express.json())
+app.use(express.json());
```
Comment (pippaamy) [inline db/app.js:11]: do we need this?

File: db/controllers/articles-controller.js [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+  const { article_id } = req.params;
+  return selectArticleById(article_id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch(next);
+};
+
+exports.getArticles = (req, res, next) => {
+  return selectArticles().then((articles) => {
```
Comment (pippaamy) [inline db/controllers/articles-controller.js:16]: lovely!

File: db/models/articles-model.js [hunk 4]
```diff
@@ -3,10 +3,29 @@ const db = require("../connection");
...
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      }
       return rows[0];
     });
 };
+
+exports.selectArticles = () => {
+  return db
+    .query(
+      `SELECT articles.author,title,articles.article_id,topic,articles.created_at,articles.votes,article_img_url, COUNT(comment_id) AS comment_count
```
Comment (pippaamy) [inline db/models/articles-model.js:17]: what is the count here doing if we are doing it below?

File: db/models/articles-model.js [hunk 5]
```diff
@@ -3,10 +3,29 @@ const db = require("../connection");
...
+    FROM articles 
+    LEFT JOIN comments 
+    ON articles.article_id = comments.article_id 
+    GROUP BY
+    articles.article_id
+    ORDER BY articles.created_at DESC;`
+    )
+    .then(({ rows }) => {
+      rows.forEach((article) => {
+        article.comment_count = Number(article.comment_count);
```
Comment (pippaamy) [inline db/models/articles-model.js:27]: we should be able to do this in the sql query :)

**Review by pippaamy — COMMENTED** [review]
Great job :) left some suggestions in the comments :)

File: db/models/articles-model.js (hunk 5 above)
Comment (AkivaKn) [reply db/models/articles-model.js:27]: The query returns with the count as a string e.g. '2'

**Pull request merged**